src/bootloaders/syslinux.c

```diff
@@ -172,19 +172,17 @@ static bool syslinux_set_default_kernel(const BootManager *manager, const Kernel
                                 abort();
                         }
                 }
-                if (streq(k->path, kernel->path)) {
```

Comment:
Test for kernel not needed, kernel is checked outside the for loop and isn't changed.